pkg/ddc/base/portallocator/port_allocator_test.go

```diff
@@ -69,6 +69,36 @@ func TestRuntimePortAllocator(t *testing.T) {
 	}
 }
 
+func TestRandomRuntimePortAllocator(t *testing.T) {
+	pr := net.ParsePortRangeOrDie("20000-21000")
+	SetupRuntimePortAllocatorWithType(nil, pr, Random, dummy)
+
+	allocator, err := GetRuntimePortAllocator()
+	if err != nil {
+		t.Errorf("get non-nil err when GetRuntimePortAllocator")
+		return
+	}
+
+	allocatedPorts, err := allocator.GetAvailablePorts(pr.Size)
```

Comment:
Since each runtime demands a group of ports by calling `allocator.GetAvailablePorts(portNum int)`, I think it necessary to ensure that all the ports returned should be unique.

Reply:
fixed and add test.